lib/init.coffee

```diff
@@ -6,11 +8,64 @@ module.exports =
       title: 'golint Executable Path'
       description: 'The path where `golint` is located'
       type: 'string'
-      default: ''
+      default: 'golint'
+
     extraOptions:
       title: 'Extra Options'
       description: 'Options for `golint` command'
       type: 'string'
       default: '-min_confidence=0.8'
+
   activate: ->
-    console.log 'activate linter-golinter'
+    @subscriptions = new CompositeDisposable
+
+    @subscriptions.add atom.config.observe 'linter-golinter.executablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+
+    @subscriptions.add atom.config.observe 'linter-golinter.extraOptions',
+      (extraOptions) =>
+        @extraOptions = extraOptions
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    provider =
+      grammarScopes: ['source.go']
+      scope: 'file'
+      lintOnFly: false
+
+      lint: (textEditor) =>
+        return new Promise (resolve, reject) =>
+          filePath = textEditor.getPath()
+          errors = []
+
+          process = new BufferedProcess
```

Comment:
You might wanna use the `atom-linter` npm module, it has some cool methods like `exec` and `parse` as well as `parseNode`